--- a/crates/deka_ui_macros/src/lib.rs
+++ b/crates/deka_ui_macros/src/lib.rs
@@ -148,7 +148,7 @@
                     quote_spanned!(name.span()=> .children(Box::new(move || ::deka_ui::View::fragment(vec![#(#children),*]))))
                 };
                 Ok(
-                    quote_spanned!(name.span()=> #name(::deka_ui::props_for(#name) #(#setters)* #children .build())),
+                    quote_spanned!(Span::call_site()=> #name(::deka_ui::props_for(#name) #(#setters)* #children .build())),
                 )
             } else {
                 Ok(
