--- a/crates/deka_ui_macros/src/lib.rs
+++ b/crates/deka_ui_macros/src/lib.rs
@@ -308,8 +308,8 @@
                     return Err(meta.error("expected one prop default"));
                 }
                 default = Some(if meta.input.peek(syn::Token![=]) {
-                    let value: Expr = meta.value()?.parse()?;
-                    quote!(#value)
+                    let _value: Expr = meta.value()?.parse()?;
+                    quote!(::core::default::Default::default())
                 } else {
                     quote!(::core::default::Default::default())
                 });
